src/main/java/com/vinyl/service/EmailValidator.java

```diff
@@ -3,26 +3,23 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import org.springframework.stereotype.Component;
-
 import com.vinyl.model.User;
 import com.vinyl.service.exception.BadRequestException;
 
-@Component
-public class EmailValidator {
+public class EmailValidator implements Validator {
```

Comment:
update names of 2 validator classes to specify that they refer to a user, UserEmailValidator, etc

Reply:
fixed